torch/_inductor/codegen/cpp.py

```diff
@@ -2986,11 +2986,13 @@ def store_reduction(self, name, index, value):
         index = self.rename_indexing(index)
         var = self.args.output(name)
         out_dtype = V.graph.get_dtype(name)
-        dtype = (
-            (out_dtype if out_dtype == torch.double else torch.float)
-            if out_dtype.is_floating_point
-            else torch.int64
-        )
+        if out_dtype.is_floating_point:
+            if out_dtype == torch.double:
+                dtype = out_dtype
+            else:
+                dtype = torch.float
+        else:
+            dtype = out_dtype
```

Comment:
You may simplify the code as below:
```suggestion
        if out_dtype.is_floating_point and out_dtype != torch.double:
            dtype = torch.float
        else:
            dtype = out_dtype
```